### PR TITLE
Driver Page

### PR DESCRIPTION
**Pull request by adityapatil834**
Drivers Page added with Shift table. 
The link to the driver page will only appear if the user has a driver role. (Admin or not doesn't matter)

Local Host Screenshot "By Mohamed Elfouly"
image

### CODE REVIEW

File: frontend/src/tests/pages/DriverPage.test.js
Comment (Mohamed-Elfouly) [inline frontend/src/tests/pages/DriverPage.test.js:61]: You might want to remove this comment

**Review by Mohamed-Elfouly — DISMISSED** [review]
LGTM, look at comment though

**Comment by Mohamed-Elfouly** [comment]
Provide description. VERY vague

**Review by cjyu81 — APPROVED** [review]
Changes looks good. The webapp works on my local machine:
image
Currently, only users with the driver role can see the drivers page. This is consistent with the acceptance criteria but you probably want to clarify this in the Release Notes or PR description. I think one would normally assume the Admin role alone would be enough to access every page. Atleast, I was confused because I had to go give myself the driver role in the admin page before the Driver page appeared.
Also please add more description to you PR, especially since this PR is very large.

File: frontend/src/App.js
Comment (chriscerie) [inline frontend/src/App.js:13]:
```suggestion
```

File: frontend/src/main/components/Driver/DriverShiftsTable.js
Comment (chriscerie) [inline frontend/src/main/components/Driver/DriverShiftsTable.js:28]: Define this outside the component to not ruin table's shallow equality optimizations.
```suggestion
const columns = [
    {
        Header: 'id',
        accessor: 'id', // accessor is the "key" in the data
    },
    {
        Header: 'Day',
        accessor: 'day',
    },
    {
        Header: 'Shift start',
        accessor: 'shiftStart',
    },
    {
        Header: 'Shift end',
        accessor: 'shiftEnd',
    },
    {
        Header: 'Backup driver',
        accessor: 'driverBackupID',
    }
];

export default function DriverShiftsTable({ drivershifts }) {
```

File: frontend/src/tests/pages/DriverPage.test.js
Comment (chriscerie) [reply frontend/src/tests/pages/DriverPage.test.js:61]: Yes, this isn't needed.

**Review by chriscerie — CHANGES_REQUESTED** [review]
@Mohamed-Elfouly @cjyu81 Approving a pr that isn't ready to be reviewed isn't good. If a pr still needs an important element like a description or meaningful title, don't be afraid to request changes without approving.

**Review by abcdefucsb — APPROVED** [review]
This pr has been code review by others. LGTM!

**Review by chriscerie — CHANGES_REQUESTED** [review]
This still needs a meaningful title & description.

**Comment by pconrad** [comment]
@adityapatil834 That's a completely unacceptable title, and it's been pointed out multiple times.  At this point, for you to not have already fixed it really sends a message.   I'll leave for you to work out what that message is, but I'll give you a clue: it isn't good.

**Comment by pconrad** [comment]
@cjyu81 and @Mohamed-Elfouly ... the right thing to do here would have been to either (a) not approve the pull request until the title was fixed, or (b) fix the title yourself, then approve it.

Approving it with a title this bad does NO one on the team any favors.

**Comment by pconrad** [comment]
Thanks for the new title. 

For reference: https://ucsb-cs156.github.io/topics/pull_requests.html#pr-titles

**Review by Mohamed-Elfouly — DISMISSED** [review]
LGTM

**Review by pconrad — DISMISSED** [review]
LGTM

File: frontend/src/tests/components/Nav/AppNavbar.test.js
Comment (pconrad) [inline frontend/src/tests/components/Nav/AppNavbar.test.js:599]:
```suggestion
    test("Driver page link should appear for a user that is a driver", async () => {
```

**Review by pconrad — CHANGES_REQUESTED** [review]

File: frontend/src/tests/pages/DriverPage.test.js
Comment (pconrad) [inline frontend/src/tests/pages/DriverPage.test.js:61]:
```suggestion
```

Please remove the comment.   Third time you've been asked.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove the comment.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**
